lucene/core/src/test/org/apache/lucene/search/TestIndexSearcher.java

```diff
@@ -237,4 +241,28 @@ public void testGetSlices() throws Exception {
     service.shutdown();
     IOUtils.close(r, dir);
   }
+
+  public void testOneSegmentExecutesOnTheCallerThread() throws IOException {
+    List<LeafReaderContext> leaves = reader.leaves();
+    AtomicInteger numExecutions = new AtomicInteger(0);
+    IndexSearcher searcher = new IndexSearcher(reader, task -> {
+      numExecutions.incrementAndGet();
+      task.run();
+    }) {
+      @Override
+      protected LeafSlice[] slices(List<LeafReaderContext> leaves) {
+        ArrayList<LeafSlice> slices = new ArrayList<>();
+        for (LeafReaderContext ctx : leaves) {
+          slices.add(new LeafSlice(Arrays.asList(ctx)));
+        }
+        return slices.toArray(new LeafSlice[0]);
+      }
+    };
+    searcher.search(new MatchAllDocsQuery(), 10);
+    if (leaves.size() <= 1) {
```

Comment:
Should this be just == 1?

Reply:
it's in theory possible to have 0 leaves. That's why I did that